hpx/util/plugin/detail/dll_dlopen.hpp

```diff
@@ -307,13 +312,32 @@ namespace hpx { namespace util { namespace plugin {
                     "plugin::get_directory",
                     str.str());
             }
+            result = directory;
+            ::dlerror();                // Clear the error state.
+#elif defined(__APPLE__)
+            // SO staticfloat's solution
+            const_cast<dll&>(*this).LoadLibrary(ec);
+            if (ec)
+            {
+                // iterate through all images currently in memory
+                for (boost::int32_t i = ::_dyld_image_count(); i >= 0; --i)
+                {
+                    const char* image_name = ::_dyld_get_image_name(i);
```

Comment:
The docs say that `_dyld_get_image_name()` can return NULL. Should this be handled gracefully?